cocos/asset/asset-manager/bin-package-unpack.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ Copyright (c) 2025 Xiamen Yaji Software Co., Ltd.
+
+ https://www.cocos.com/
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights to
+ use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ of the Software, and to permit persons to whom the Software is furnished to do so,
+ subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in
+ all copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+*/
+
+/**
+ * bin package format example.
+|PACK_BIN_TYPE - 4bytes|
+|VERSION - 4bytes|
+|FILES_COUNT - 4bytes|
+|FILE_1_SIZE - 4bytes|
+|FILE_2_SIZE - 4bytes|
+...
+|FILE_N_SIZE - 4bytes|
+|PACKED_BIN|
+ */
+
+const PACK_BIN_TYPE = 'BINP';
+const VERSION = 1;
+const UNIT_SIZE = 4;
+const LITTLE_ENDIAN = true;
+
+export function binPackageUnpack (arrayBuffer: ArrayBuffer): ArrayBuffer[] {
+    const dataView = new DataView(arrayBuffer, 0, arrayBuffer.byteLength);
+
+    const packBinType = String.fromCharCode(...new Uint8Array(arrayBuffer.slice(0, UNIT_SIZE)));
+    const version = dataView.getUint32(UNIT_SIZE, LITTLE_ENDIAN);
+    if (packBinType !== PACK_BIN_TYPE || version !== VERSION) {
+        throw new Error('Invalid bin package format');
+    }
+
+    const filesCount = dataView.getUint32(UNIT_SIZE * 2, LITTLE_ENDIAN);
+    const filesPosition: {offset: number, size: number}[] = [];
+
+    let offset = UNIT_SIZE * (3 + filesCount);
+    for (let i = 0; i < filesCount; i++) {
+        const size = dataView.getUint32(UNIT_SIZE * (3 + i), LITTLE_ENDIAN);
+        filesPosition.push({ offset, size });
+        offset += size;
+    }
+
+    return filesPosition.map(({ offset, size }) => {
+        const buffer = arrayBuffer.slice(offset, offset + size);
```

Comment:
arrayBuffer.slice will do a copy operation.
I think we could return `Uint8Array` which is ArrayBufferView for  `binPackageUnpack` function.
All ArrayBufferViews share the same underlying ArrayBuffer.